datashader/data_libraries/dask.py

```diff
@@ -27,6 +35,7 @@ def dask_pipeline(df, schema, canvas, glyph, summary, *, antialias=False, cuda=F
     if isinstance(dsk, da.Array):
         return da.compute(dsk, scheduler=scheduler)[0]
 
+    df = getattr(df, 'optimize', lambda: df)()  # See note at the top of this file
```

Comment:
Only suggestion I'd have would be to encapsulate this in a `dask_compat` function so there's a clearer place to put the docstring.